src/QMCWaveFunctions/Fermion/MultiSlaterDetTableMethod.h

```diff
@@ -285,8 +285,10 @@ class MultiSlaterDetTableMethod : public WaveFunctionComponent, public Optimizab
   std::shared_ptr<std::vector<ValueType>> C;
   /// if true, the CI coefficients are optimized
   bool CI_Optimizable;
+public:
   //optimizable variable is shared with the clones
   std::shared_ptr<opt_variables_type> myVars;
+private:
```

Comment:
This is a big no. Functions like evaluateDerivativesWF do not expect the class member myVars being passed in as an argument as well. Treat this UB and bug. Please make a myVar object at the consumer side.

Reply:
Agreeing with Ye’s comment.

Breaking encapsulation by making myVars public screams that, at a minimum, we are missing some accessor functions, or the design needs revision. Perhaps both. Recently it has not taken much work to avoid making variables public. If it is not clear how to avoid this we should discuss.

(Unfortunately the oldest code in QMCPACK has numerous similar problems that we are still dealing with.)

Reply:
For now, I will remove this code locally and put in a non-implemented runtime error.  I will be handling this and related code in a future PR, but will need access to the appropriate myVars for it (which currently resides at driver level, I believe)

Reply:
Resolved.